src/x509.rs

```diff
@@ -115,7 +115,7 @@ impl<'a> AttributeTypeAndValue<'a> {
 
     /// Get the content as a slice.
     #[inline]
-    pub fn as_slice(&'a self) -> &'a [u8] {
+    pub fn as_slice(&self) -> &'a [u8] {
```

Comment:
This does not look correct. By forcing only the return to satisfy `'a`, the compiler assigns an unnamed lifetime to `self`, and they do not match. Removing the return lifetime fixes that.
I'll add some commits.

Reply:
This is not the way lifetimes work, and your reversal of this change partially obviates the intended consequences of this PR. Please see my comment in https://github.com/rusticata/asn1-rs/pull/22#issuecomment-1507155046 where I address your misconceptions about lifetimes.

> By forcing only the return to satisfy 'a, the compiler assigns an unnamed lifetime to self, and they do not match.

You seem think that "they do not match" is a problematic statement, but in fact, that's quite literally the point here: we _want_ the receiver's lifetime (`&self`) to differ from the returned lifetime (`&[u8]`). Otherwise, we ask for unification, which means that the receiver must live as long as the return value and vice-versa, even when no such relationship is needed.

Your commit has obviated the meaningful change here and resulted in a strictly worse situation wherein the content of the struct is typed to a particular reference (the receiver, created exactly when the method is called - it doesn't exist otherwise) to the struct. This only makes sense when the struct owns the data, which is not the case here.